### PR TITLE
Use Liste of Mirrors Urls for Mangalek "ar"

### LINKED ISSUE #1054: MangaLike : Source request
### Source name

MangaLike

### Source link

https://manga-like.org/

### Source language

Arabic

### Other details

_No response_

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #2039: Mangalek : Moved to multiplayer section
### Source name

MangaLek

### Source language

Arabic

### Describe your suggested feature

The source is moved to the multiple section and all its sources are added under one source, the multiple section, so that it contains all of its multiple sources and their locations, and these are its sources :
1- MangaLek
2- MangaLink
3- MangaStarz
4- MangaSpark
5- MangaLionz 

### Other details

This is an example picture
 
Screenshot_٢٠٢٤٠٣٢٢-١٧٠٧٢٩_Mihon

If this feature is possible, it would be very wonderful to add it, as all the sources are one company with different names

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by mohamedotaku**
Close #1054 
Close #2039 

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt [hunk 1]
```diff
@@ -35,38 +34,46 @@ class Mangalek :
...
+        private const val MIRROR_PREF_KEY = "MIRROR"
+        private const val MIRROR_PREF_TITLE = "تعديل رابط مانجا ليك"
+        internal val MIRROR_PREF_ENTRY_VALUES = arrayOf(
+            "https://lekmanga.net",
+            "https://manga-lek.org",
+            "https://like-manga.net",
+            "https://lekmanga.com",
+        )
+        private val MIRROR_PREF_DEFAULT_VALUE = MIRROR_PREF_ENTRY_VALUES[0]
+        private const val RESTART_TACHIYOMI = ".لتطبيق الإعدادات الجديدة Tachiyomi أعد تشغيل"
```
Comment (AwkwardPeak7) [inline src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt]: move these out of the class and use `MIRROR_PREF_DEFAULT_VALUE ` in the constructor as well

File: src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt [hunk 2]
```diff
@@ -35,38 +34,46 @@ class Mangalek :
...
+            entries = MIRROR_PREF_ENTRY_VALUES
+            entryValues = MIRROR_PREF_ENTRY_VALUES
+            setDefaultValue(MIRROR_PREF_DEFAULT_VALUE)
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                preferences.edit().putString(MIRROR_PREF_KEY, entry).apply()
```
Comment (AwkwardPeak7) [inline src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt]: shouldn't be necessary

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt (hunk 1 above)
Comment (mohamedotaku) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt]: Ok

File: src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt [hunk 3]
```diff
@@ -35,38 +45,30 @@ class Mangalek :
     override val useLoadMoreRequest = LoadMoreStrategy.Always
     override val chapterUrlSuffix = ""
 
-    private val defaultBaseUrl = mangalekUrl
-    override val baseUrl by lazy { getPrefBaseUrl() }
+    override val baseUrl by lazy { getMirrorPref() }
```
Comment (AwkwardPeak7) [inline src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt]: make it something like [this](https://github.com/keiyoushi/extensions-source/blob/59c3a8d9acde0e967d31320ea93860bfa49bb542/src/vi/ungtycomics/src/eu/kanade/tachiyomi/extension/vi/ungtycomics/UngTyComics.kt#L31)

This is to make sure the automod on github is able detect all the mirrors for the site

Comment (mohamedotaku) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt]: Ook

Comment (vetleledaal) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt]: "CI" is not a country code, and should not be set to "AR". CI stands for Continuous Integration, which is used to detect the mirrors.

File: src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt [hunk 4]
```diff
@@ -21,11 +21,21 @@ import uy.kohesive.injekt.api.get
...
+private val MIRROR_PREF_DEFAULT_VALUE = MIRROR_PREF_ENTRY_VALUES[0]
+private const val RESTART_TACHIYOMI = ".لتطبيق الإعدادات الجديدة Tachiyomi أعد تشغيل"
+
 class Mangalek :
     Madara(
         "مانجا ليك",
-        mangalekUrl,
+        MIRROR_PREF_DEFAULT_VALUE,
         "ar",
         SimpleDateFormat("MMMM dd, yyyy", Locale("ar")),
```
Comment (vetleledaal) [inline src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt:41]: Doesn't work for `lekmanga.org`, which uses `yyyy-MM-dd`, not sure if there's a decent approach to handling that.

File: src/ar/mangalek/build.gradle [hunk 5]
```diff
@@ -3,7 +3,7 @@ ext {
     extClass = '.Mangalek'
     themePkg = 'madara'
     baseUrl = 'https://lekmanga.net'
-    overrideVersionCode = 5
+    overrideVersionCode = 6
```
Comment (vetleledaal) [inline src/ar/mangalek/build.gradle:6]: should set `isNsfw`. I didn't find any, if you can't find any as well use:
```suggestion
    overrideVersionCode = 6
    isNsfw = false
```

File: src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt (hunk 4 above)
Comment (AwkwardPeak7) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt:41]: can override `parseChapterDate` from Madara and try both formats; if first results in 0, try second one

Comment (mohamedotaku) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt:41]: I have no idea how to fix it, i tried many solutions but that's not work.

Comment (AwkwardPeak7) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt:41]:
```
private val formatOne = SimpleDateFormat("MMMM dd, yyyy", Locale("ar"))
private val formatTwo = SimpleDateFormat("yyyy-MM-dd", Locale.US)

override fun parseChapterDate(date: String?): Long {
    date ?: return 0L

    return try {
        formatOne.parse(date)!!.time
    } catch (_: ParseException) {
        try {
        	formatTwo.parse(date)!!.time
        } catch (_: ParseException) {
            0L
        }
    }
}
```
try this

Comment (mohamedotaku) [reply src/ar/mangalek/src/eu/kanade/tachiyomi/extension/ar/mangalek/Mangalek.kt:41]: > ```
> private val formatOne = SimpleDateFormat("MMMM dd, yyyy", Locale("ar"))
> private val formatTwo = SimpleDateFormat("yyyy-MM-dd", Locale.US)
> 
> override fun parseChapterDate(date: String?): Long {
>     date ?: return 0L
> 
>     return try {
>         formatOne.parse(date)!!.time
>     } catch (_: ParseException) {
>         try {
>         	formatTwo.parse(date)!!.time
>         } catch (_: ParseException) {
>             0L
>         }
>     }
> }
> ```
> 
> try this

ok, thank you

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**